lib/sequence.js

```diff
@@ -167,6 +167,52 @@ Sequence.prototype._setHooks = function() {
             done();
         }.bind(doc));
     });
+
+    this._schema.pre('findOneAndUpdate', function(next) {
+        // jscs:disable
+        var _thisquery = this,
+            referenceValue;
+
+        // jscs:enable
+
+        this._update['$setOnInsert'] = this._update['$setOnInsert'] || {};
+
+        this.model.findOne({_id: _thisquery._update._id}, function(err, obj) {
```

Comment:
Afaik, mongoose documents always have _id fields, so if the query doesn't have it, then it should be a new document?